autoload/slime.vim

```diff
@@ -46,15 +46,17 @@ endfunction
 
 function! s:KittySend(config, text)
   call s:WritePasteFile(a:text)
-  call system("kitty @ send-text --match id:" . shellescape(a:config["window_id"]) .
+  call system("kitty @ --to " . shellescape(a:config["listen_on"]) .
+    \ " send-text --match id:" . shellescape(a:config["window_id"]) .
     \ " --from-file " . g:slime_paste_file)
 endfunction
 
 function! s:KittyConfig() abort
   if !exists("b:slime_config")
-    let b:slime_config = {"window_id": 1}
+    let b:slime_config = {"window_id": 1, "listen_on": $KITTY_LISTEN_ON}
```

Comment:
I have mixed feelings about this … now everybody needs to specify a `listen_on` 😩 

maybe we could leave it blank and skip the `--to` when it's blank? (is that better or more confusing?)

Reply:
Would it work to simply add a `if has('nvim')` condition in order not to force regular vim users to specify a listen_on for vim-slime to work in kitty ?

Reply:
i guess if you had multiple kitty sessions open, and wanted to slime it to a different session, the override would be helpful, but otherwise hitting enter through the second input is what i think most people would do

if we decide that overriding is sufficiently unlikely that we just default to `KITTY_LISTEN_ON`, in the case where you're in nvim and `$KITTY_LISTEN_ON` is empty it could also just error

at any rate i put the nvim conditional on it out of laziness, but am open to follow ups if folx feel strongly about it

Reply:
I got burned myself on:
- missing `allow_remote_control=yes`
- calling `kitty @ send-text` from iterm2 ("outside" kitty)

so, I don't think it's just an nvim problem.

I think this PR makes things better, but I'll add to it